File: problems/Top K Frequent Words/topKFrequent.cs
public class Solution {
    public IList<string> TopKFrequent(string[] words, int k) {
        // loop through to find occurances
        // store the pair of word(K) and freq(V) in to map and the words in a list
		// sort the words using comparator
        var map = new Dictionary<string, int>();
        var allwords = new List<string>();

        foreach (string word in words)
        {
            if (!map.ContainsKey(word))
            {
                map.Add(word, 0);
                allwords.Add(word);
            }

            map[word]++;
        }

        // sort the array based on map
        allwords.Sort((x, y) => map[x].CompareTo(map[y]) != 0 ? map[y].CompareTo(map[x]) : x.CompareTo(y));
        return allwords.Take(k).ToList();
    }
}

File: problems/Top K Frequent Words/topKFrequent.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        List<String> result = new ArrayList<String>();
        HashMap<String, Integer> wordFrequenties = new HashMap<String, Integer>();
        PriorityQueue<String> minHeap = new PriorityQueue<String>(
            (a,b) -> wordFrequenties.get(a) == wordFrequenties.get(b)
            ? b.compareTo(a)
            : wordFrequenties.get(a) - wordFrequenties.get(b)
        );

        for (String word : words) {
            wordFrequenties.put(word, 1 + wordFrequenties.getOrDefault(word, 0));
        }

        for (String key : wordFrequenties.keySet()) {
            minHeap.add(key);

            if (k < minHeap.size()) {
                minHeap.poll();
            }
        }

        while (!minHeap.isEmpty()) {
            result.add(0, minHeap.poll());
        }

        return result;
    }
}
